# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/HTTPSSVC.cpp
# Commit: 26c5be98ea21
# Full Hash: 26c5be98ea21c8c331cefe9649e3eb53197cd821
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-08-26 03:24:37
# Regressor Bug: 1652667
# File Overlap Count: 1
# Description:
#   Bug 1652667 - P2: Telemetry for whether a connection is succeeded by using an HTTPSSVC record r=valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D87857
# ==============================================================================

diff -r 96bd98319c62 -r 26c5be98ea21 netwerk/dns/HTTPSSVC.cpp
--- a/netwerk/dns/HTTPSSVC.cpp	Tue Aug 25 17:29:32 2020 +0000
+++ b/netwerk/dns/HTTPSSVC.cpp	Tue Aug 25 17:29:31 2020 +0000
@@ -196,9 +196,12 @@
 
 already_AddRefed<nsISVCBRecord>
 DNSHTTPSSVCRecordBase::GetServiceModeRecordInternal(
-    bool aNoHttp2, bool aNoHttp3, const nsTArray<SVCB>& aRecords) {
+    bool aNoHttp2, bool aNoHttp3, const nsTArray<SVCB>& aRecords,
+    bool& aRecordsAllExcluded) {
   nsCOMPtr<nsISVCBRecord> selectedRecord;
   uint32_t recordHasNoDefaultAlpnCount = 0;
+  uint32_t recordExcludedCount = 0;
+  aRecordsAllExcluded = false;
   nsCOMPtr<nsIDNSService> dns = do_GetService(NS_DNSSERVICE_CONTRACTID);
   for (const SVCB& record : aRecords) {
     if (record.mSvcFieldPriority == 0) {
@@ -215,6 +218,7 @@
                                                 &excluded)) &&
         excluded) {
       // Skip if the domain name of this record was failed to connect before.
+      ++recordExcludedCount;
       continue;
     }
 
@@ -240,6 +244,10 @@
     return nullptr;
   }
 
+  if (recordExcludedCount == aRecords.Length()) {
+    aRecordsAllExcluded = true;
+  }
+
   return selectedRecord.forget();
 }
 